src/Generator/Check24DE.php

```diff
@@ -3,164 +3,349 @@
+    {
+        // get the price
+        $priceList = $this->elasticExportPriceHelper->getPriceList($variation, $settings);
+
+        // only variations with the Retail Price greater than zero will be handled
+        if(!is_null($priceList['price']) && $priceList['price'] > 0)
+        {
+            $variationName = $this->elasticExportHelper->getAttributeValueSetShortFrontendName($variation, $settings);
+
+            $shippingCost = $this->getShippingCost($variation);
+
+            $manufacturer = $this->getManufacturer($variation);
+
+            $stock = $this->stockHelper->getStock($variation);
+
+            $price['variationRetailPrice.price'] = $priceList['price'];
+
+            $data = [
+                'id' 				=> $this->getSku($variation),
+                'manufacturer' 		=> $manufacturer,
+                'mpnr' 				=> $variation['data']['variation']['model'],
+                'ean' 				=> $this->elasticExportHelper->getBarcodeByType($variation, $settings->get('barcode')),
+                'name' 				=> $this->elasticExportHelper->getMutatedName($variation, $settings) . (strlen($variationName) ? ' ' . $variationName : ''),
+                'description' 		=> $this->elasticExportHelper->getMutatedDescription($variation, $settings),
+                'category_path' 	=> $this->elasticExportHelper->getCategory((int)$variation['data']['defaultCategories'][0]['id'], $settings->get('lang'), $settings->get('plentyId')),
+                'price' 			=> $priceList['price'],
+                'price_per_unit'	=> $this->elasticExportHelper->getBasePrice($variation, $price, $settings->get('lang')),
+                'link' 				=> $this->elasticExportHelper->getMutatedUrl($variation, $settings, true, false),
+                'image_url'			=> $this->elasticExportHelper->getMainImage($variation, $settings),
+                'delivery_time' 	=> $this->elasticExportHelper->getAvailability($variation, $settings, false),
+                'delivery_cost' 	=> $shippingCost,
+                'pzn' 				=> '',
+                'stock' 			=> $stock,
+                'weight' 			=> $variation['data']['variation']['weightG']
+            ];
+
+            $this->addCSVContent(array_values($data));
+        }
+        else
+        {
+            $this->getLogger(__METHOD__)->info('ElasticExportCheck24DE::logs.variationNotPartOfExportPrice', [
+                'VariationId' => $variation['id']
+            ]);
         }
     }
 
     /**
      * Get the SKU.
-     * @param  array $item
+     *
+     * @param  array $variation
      * @return string
      */
-    private function getSku($item):string
+    private function getSku($variation):string
     {
-        return (string) $this->elasticExportHelper->generateSku($item['id'], self::CHECK24_DE, 0, (string)$item['data']['skus']['sku']);
+        if(!is_null($variation['data']['skus'][0]['sku']) && strlen($variation['data']['skus'][0]['sku']) > 0)
```

Comment:
@mbrueschke Is that here not the same problem you had in the past which the index access to the array?